networkit/cpp/distance/SPSP.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPSP.cpp
+ *
+ *  Created on: 23.10.2020
+ *      Author: Eugenio Angriman <angrimae@hu-berlin.de>
+ */
+
+// networkit-format
+
+#include <memory>
+#include <omp.h>
+
+#include <networkit/distance/BFS.hpp>
+#include <networkit/distance/Dijkstra.hpp>
+#include <networkit/distance/SPSP.hpp>
+#include <networkit/distance/SSSP.hpp>
+
+namespace NetworKit {
+
+void SPSP::run() {
+    distances.clear();
+    distances.resize(sources.size(), std::vector<edgeweight>(G->upperNodeIdBound()));
```

Comment:
Wouldn't it be more efficient to initialize the distances with empty vectors? Or is there any advantage in sequentially allocating the memory?

Reply:
True, the distance vectors are already allocated in `SSSP`, allocating memory here is useless.